examples/read_pem_files/main.rs

```diff
@@ -0,0 +1,69 @@
+use config::{Config, Format, Map, Value, ValueKind, FileStoredFormat};
+use std::io::{Error, ErrorKind};
+
+/// The private and public key sources will be read into their associated variable:
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct Settings {
+    pub private_key: Option<String>,
+    pub public_key: Option<String>,
+}
+
+fn main() {
+    // Sourcing from two separate files for the `Settings` struct,
+    // these are both marked as optional to avoid failure if the file doesn't exist.
+    let file_public_key = config::File::new("public.pem", PemFile).required(false);
```

Comment:
This has to be changed to final destination of the files

Reply:
Ah yes sorry, this is my fault as I tested it locally on a new project with the current `config-rs` crate installed as if a user was applying the example, rather than running from the crates examples 😅

Reply:
From [comment](https://github.com/mehcode/config-rs/pull/479#issuecomment-1773884294):

> _I agree with you, perhaps we should rename the example to `custom_file` or `custom_format_from_file`._
> _The current `custom_format` example is only applicable to strings, then I think that the main purpose of this should be just an example to demonstrate how to open custom files and use `config::File::new()` feature._
> _About folder name, we could rename it to files, it fits better its purpose._

Yeah that sounds good. Let's go with `custom_file_format`? (_If you think `custom_format_from_file` seems more appropriate, feel free to apply that change_)

`files/` for the folder sounds good too 👍 

```suggestion
    let file_public_key = config::File::new("examples/custom_file_format/files/public.pem", PemFile).required(false);
```